evalml/pipelines/binary_classification_pipeline.py

```diff
@@ -60,3 +69,34 @@ def _score(X, y, predictions, objective):
         if predictions.ndim > 1:
             predictions = predictions.iloc[:, 1]
         return ClassificationPipeline._score(X, y, predictions, objective)
+
+    def _compute_predictions(self, X, y, objectives, time_series=False):
+        """Compute predictions/probabilities based on objectives."""
+        y_predicted = None
+        y_predicted_proba = None
+        if any(o.score_needs_proba for o in objectives) or (any(not o.score_needs_proba for o in objectives) and self.threshold is not None):
+            y_predicted_proba = self.predict_proba(X, y) if time_series else self.predict_proba(X)
+        if any(not o.score_needs_proba for o in objectives) and self.threshold is None:
+            y_predicted = self._predict(X, y, pad=True) if time_series else self._predict(X)
```

Comment:
Is this pad something that we always want on or that the user would always expect to be True?

Reply:
I think for time-series, yes! @freddyaboulton correct me if I'm wrong 😅

Reply:
Yes! It's an internal detail of time series pipelines but we always want the predicted probabilities and predictions to be padded. @angela97lin Can you add coverage for time series pipelines to `test_binary_predict_pipeline_use_objective`?